libs/agno/agno/db/postgres/postgres.py

```diff
@@ -348,6 +363,7 @@ def _upsert_team_session_raw(self, session: TeamSession) -> Optional[dict]:
             )
             runs = [run.to_dict() for run in session.runs] if session.runs else None
             summary = session.summary if session.summary else None
+            summary = summary.to_dict() if isinstance(summary, SessionSummary) else summary
```

Comment:
Shouldn't all the above be part of the session.to_dict()? Otherwise you have to do this in each DB type...

Reply:
Yep good one. Agree the model should be responsible for parsing itself. I'll refactor this everywhere